src/main/java/com/google/devtools/build/lib/bazel/bzlmod/SingleExtensionEvalFunction.java

```diff
@@ -342,10 +342,17 @@ public String getRecordedDiffMessages() {
   private static boolean didRepoMappingsChange(
       Environment env, ImmutableTable<RepositoryName, String, RepositoryName> recordedRepoMappings)
       throws InterruptedException, NeedsSkyframeRestartException {
+    // Request repo mappings for any 'source repos' in the recorded mapping entries.
+    // Note specially that the main repo needs to be treated differently: if any .bzl file from the
+    // main repo was used for module extension eval, it _has_ to be before WORKSPACE is evaluated,
```

Comment:
Could you link to the relevant BzlLoadFunction part from here as well? That's what made it clear for me why we don't have to look into individual WORKSPACE chunks' repo mappings.

Reply:
I'm usually reluctant to link to specific locations in code (unless it's with JavaDoc `{@link }`) because they tend to get out of date. But I added a small reference to the method in question.